pkg/systray/runtime/runner_darwin.go

```diff
@@ -0,0 +1,179 @@
+//go:build darwin
+// +build darwin
+
+package runtime
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"strconv"
+	"syscall"
+
+	"github.com/go-kit/kit/log/level"
+)
+
+func (r *SystrayUsersProcessesRunner) runConsoleUserSystray() error {
+	consoleOwnerUid, err := consoleOwnerUid()
+	if err != nil {
+		return fmt.Errorf("getting console owner uid: %w", err)
+	}
+
+	// there seems to be a brief moment during start up where root or system (non-human)
+	// users own the console, if we spin up the process for them it will add an
+	// unnecessary process. On macOS human users start at 501
+	if consoleOwnerUid < 500 {
+		level.Debug(r.logger).Log(
+			"msg", "skipping systray for root or system user",
+			"uid", consoleOwnerUid,
+		)
+
+		return nil
+	}
+
+	uid := fmt.Sprint(consoleOwnerUid)
+
+	// already have a systray for the console owner
+	if proc, ok := r.uidProcs[uid]; ok {
+		// if the process is still running, return
+		if processExists(proc.Pid) {
+			return nil
+		}
+
+		// proc is dead
+		level.Info(r.logger).Log(
+			"msg", "existing systray process dead for console user, starting new systray process",
+			"dead_pid", r.uidProcs[uid].Pid,
+			"uid", consoleOwnerUid,
+		)
+	}
+
+	executable, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("getting executable path: %w", err)
+	}
+
+	proc, err := runAsUser(uid, executable, "systray")
+	if err != nil {
+		return fmt.Errorf("running systray: %w", err)
+	}
+	r.uidProcs[uid] = proc
+
+	level.Debug(r.logger).Log(
+		"msg", "systray started",
+		"uid", consoleOwnerUid,
+		"pid", proc.Pid,
+	)
+
+	return nil
+}
+
+func consoleOwnerUid() (uint32, error) {
```

Comment:
Seems fine for now, but if I think about making this multi-platform, I wonder if we'd want to move the core logic into a common routine, and make a per platform `consoleUser` function. TBD I guess

Reply:
once we add functionality for other OSes I think we'll get a clearer picture, for now I just took the quickest, least amount of place holder functions I could.